cmd/osbuild-composer/composer.go

```diff
@@ -307,14 +311,49 @@ func (c *Composer) Start() error {
 	if c.weldrListener != nil {
 		go func() {
 			err := c.weldr.Serve(c.weldrListener)
-			if err != nil {
+			if err != nil && err != http.ErrServerClosed {
 				panic(err)
 			}
 		}()
 	}
 
-	// wait indefinitely
-	select {}
+	sigint := make(chan os.Signal, 1)
+
+	signal.Notify(sigint, syscall.SIGTERM)
+	signal.Notify(sigint, syscall.SIGINT)
+
+	// block until interrupted
+	<-sigint
+
+	if c.apiListener != nil {
+		err = composerAPI.Shutdown(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	if c.localWorkerListener != nil {
+		err = localWorkerAPI.Shutdown(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	if c.workerListener != nil {
+		err = remoteWorkerAPI.Shutdown(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	if c.weldrListener != nil {
+		err = c.weldr.Shutdown(context.Background())
+		if err != nil {
+			panic(err)
+		}
+	}
```

Comment:
It would be nice if we can also `Shutdown()` the `jobqueue` here. (only `dbjobqueue` implements `Shutdown` so we might want to extend this also for `fsjobqueue` as a dummy function).

Reply:
Hmm, what do you have in mind? We should shut down our external APIs, but not internal things, no?

Reply:
I did not change this.

Reply:
While I agree that this is an improvement, I think it would be nicer to shutdown all goroutines in a defined way before we exit the whole process. We can surely do it in another PR thouugh.

Reply:
Ah, yes. We should, but need to be careful about deadlocks for that (can't do it at the same time as the others). Let's do that in a follow up

Reply:
Makes sense